src/cli/subcommand/export.rs

```diff
@@ -0,0 +1,270 @@
+use std::collections::HashMap;
+use std::env;
+use std::ffi::{OsStr, OsString};
+use std::io::{stdout, Write};
+use std::os::unix::ffi::OsStrExt;
+use std::path::PathBuf;
+use std::process::ExitCode;
+
+use crate::cli::CommandExecute;
+use clap::Parser;
+
+const LOCAL_STATE_DIR: &str = "/nix/var";
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("The HOME environment variable is not set.")]
+    HomeNotSet,
+
+    #[error("__ETC_PROFILE_NIX_SOURCED is set, indicating the relevant environment variables have already been set.")]
+    AlreadyRun,
+
+    #[error("Some of the paths from Nix for XDG_DATA_DIR are not valid, due to an illegal character, like a colon.")]
```

Comment:
We should print the paths, eg

```suggestion
    #[error("The paths {} from Nix for XDG_DATA_DIR are not valid, due to an illegal character, like a colon.", 0.iter().map(|v| format!("`{}`", v.display())).join(",")]
```